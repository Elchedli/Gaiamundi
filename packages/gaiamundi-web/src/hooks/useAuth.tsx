import React from 'react';
import {
  QueryObserverResult,
  RefetchOptions,
  UseMutateFunction,
  useQuery,
  useQueryClient,
} from 'react-query';

import { Alert } from 'components/Alert/Alert';
import { LoadingMessage } from 'components/Loader/LoadingMessage';
import { User } from 'interfaces/user';
import { strapi } from '../services/strapi';
import { useLocalStorageState } from './useLocalStorageState';

// const updateAccount = async (id: number, userData: UserSignUpFields) => {
//   return (await strapi.update(ContentType.USERS, id, userData)) as User;
// };

export interface AuthContextValue {
  user: User | undefined;
  isAuthenticated: boolean;
  setUser: (data: User | undefined) => void;
  authenticate: (user: User, jwt: string) => void;
  logout: UseMutateFunction;
  refetchUser: (
    options?: RefetchOptions | undefined
  ) => Promise<QueryObserverResult<User, Error>>;
  error: Error | null;
}

const AuthContext = React.createContext<AuthContextValue | null>(null);
AuthContext.displayName = 'AuthContext';

export interface AuthProviderProps {
  children: React.ReactNode;
}

export const AuthProvider = ({ children }: AuthProviderProps): JSX.Element => {
  const { value: jwtToken, persistValue: persistJwtToken } =
    useLocalStorageState('token');
  const queryClient = useQueryClient();
  const key = 'auth-user';

  const {
    data: user,
    error,
    isLoading,
    refetch,
  } = useQuery<User, Error>({
    queryKey: key,
    queryFn: async () => {
      return await strapi.currentUser(jwtToken);
    },
    enabled: !!jwtToken,
  });

  const setUser = (data: User | undefined) =>
    queryClient.setQueryData(key, data);

  const authenticate = (user: User, jwt: string) => {
    setUser(user);
    persistJwtToken(jwt);
  };

  const logout = () => {
    persistJwtToken('');
    queryClient.clear();
  };

  if (isLoading) {
    return <LoadingMessage />;
  }

  return (
    <AuthContext.Provider
      value={{
        user,
        isAuthenticated: !!jwtToken,
        error,
        setUser,
        authenticate,
        refetchUser: refetch,
        logout,
      }}
    >
      {error && <Alert>{error.message}</Alert>}
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = () => {
  const context = React.useContext(AuthContext);
  if (!context) {
    throw new Error(`useAuth must be used within an AuthProvider`);
  }
  return context;
};
