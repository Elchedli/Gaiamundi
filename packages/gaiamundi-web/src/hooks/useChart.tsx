import { CHART_TYPES } from 'interfaces/chart';
import { useMemo } from 'react';
import { useQuery } from 'react-query';
import { getChartById } from 'services/chart';

export const useChart = (chartId: number) => {
  const query = useQuery({
    queryKey: ['chart', chartId],
    queryFn: async () => await getChartById(chartId),
    enabled: !!chartId,
  });
  const chart = query.data?.data;
  const ChartComponent = useMemo(() => {
    return chart?.type
      ? CHART_TYPES[chart.type].ChartComponent
      : CHART_TYPES.column.ChartComponent;
  }, [chart]);

  return {
    ...query,
    ChartComponent,
  };
};
