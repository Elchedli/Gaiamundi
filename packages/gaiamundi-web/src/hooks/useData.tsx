import { Alert } from 'components/Alert/Alert';
import { LoadingMessage } from 'components/Loader/LoadingMessage';
import { RawDatum, RawDatumType } from 'interfaces/chart';
import { createContext, FC, ReactNode, useContext, useMemo } from 'react';
import { useQuery } from 'react-query';
import { getPageCartoData } from 'services/dataset';
import { getDataTypeMap } from 'utils/chart';
import { calculateIndicator } from 'utils/equation';
import { usePageCarto } from './usePageCarto';

type DatasetContextType = {
  rawData: RawDatum[];
  dataKeys: Record<string, RawDatumType>;
};

const initialContext: DatasetContextType = {
  rawData: [],
  dataKeys: {},
};

const DatasetContext = createContext<DatasetContextType>(initialContext);

type DataProviderProps = {
  pageCartoId: number;
  children: ReactNode;
};

export const DataProvider: FC<DataProviderProps> = ({
  pageCartoId,
  children,
}) => {
  const { indicators } = usePageCarto();
  const {
    data: sourceData,
    error,
    isLoading,
  } = useQuery({
    queryKey: ['page-carto-dataset', pageCartoId],
    queryFn: async () => await getPageCartoData(pageCartoId),
    enabled: !!pageCartoId,
  });

  const rawData = useMemo(
    () =>
      sourceData?.map((datum) => {
        // Calculate indicators
        const computedDatum = indicators.reduce((acc, indicator) => {
          acc[indicator.name] = calculateIndicator(indicator, datum);
          return acc;
        }, {} as RawDatum);
        return {
          // Original data
          ...datum,
          // Computed data (calculated indicators)
          ...computedDatum,
        };
      }),
    [sourceData, indicators]
  );

  const dataKeys = useMemo(() => {
    if (rawData && rawData.length > 0) {
      return getDataTypeMap(rawData[0]);
    }
    return {};
  }, [rawData]);

  if (isLoading) {
    return <LoadingMessage />;
  }

  if (error) {
    return <Alert>Erreur lors du chargement des donn√©es</Alert>;
  }

  console.info('>>>>> Raw Data :', rawData);

  return (
    <DatasetContext.Provider
      value={{
        rawData: rawData || [],
        dataKeys,
      }}
    >
      {children}
    </DatasetContext.Provider>
  );
};

export const useData = () => {
  return useContext(DatasetContext);
};
