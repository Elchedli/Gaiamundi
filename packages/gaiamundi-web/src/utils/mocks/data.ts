import { ApiCollection, ApiData } from 'interfaces/api';
import { DataFragment } from 'interfaces/data-fragment';
import { Dataset } from 'interfaces/dataset';
import { UploadedFile } from 'interfaces/file';
import { GeoMap } from 'interfaces/geo-map';
import { Indicator } from 'interfaces/indicator';
import { PageCarto } from 'interfaces/page-carto';
import { Tag } from 'interfaces/tag';
import { User } from 'interfaces/user';
import mockMapJson from './map.json';

export const mockMapPath = mockMapJson;

export const mockFileUpload = {
  id: 184,
  name: 'exampleParsing2.csv',
  alternativeText: null,
  caption: null,
  width: null,
  height: null,
  formats: null,
  hash: 'example_Parsing2_ce8756a044',
  ext: '.csv',
  mime: 'text/csv',
  size: 0.05,
  url: '/uploads/example_Parsing2_ce8756a044.csv',
  previewUrl: null,
  provider: 'local',
  provider_metadata: null,
  createdAt: '2023-03-16T18:20:09.476Z',
  updatedAt: '2023-03-16T18:20:09.476Z',
};

export const mockUser: User = {
  id: 1,
  username: 'testuser',
  password: 'test',
  email: 'test@test.com',
  provider: 'local',
  confirmed: true,
  blocked: false,
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockGeoJson: ApiData<UploadedFile> = {
  id: 1,
  name: 'test.geojson',
  hash: 'abc123',
  provider: 'provider123',
  ext: '.geojson',
  mime: 'application/json',
  size: 1234,
  url: '/api/v1/geo-maps/id/thumbnail',
  formats: {
    thumbnail: {
      extension: 'ext',
      url: 'happyUrl',
      size: 360,
      hash: 'hashbar',
      mime: 'mimeExample',
    },
  },
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockCover: ApiData<UploadedFile> = {
  id: 3,
  name: 'image.png',
  hash: 'hashImage',
  provider: 'provider123',
  ext: '.png',
  mime: 'application/image',
  size: 1234,
  url: '/api/v1/geo-maps/id/thumbnailImage',
  formats: {
    small: {
      extension: 'ext',
      url: '/api/v1/geo-maps/id/thumbnailImage?720',
      size: 360,
      hash: 'hash720',
      mime: 'mimeExample',
    },
    thumbnail: {
      extension: 'ext',
      url: '/api/v1/geo-maps/id/thumbnailImage?720',
      size: 360,
      hash: 'hash720',
      mime: 'mimeExample',
    },
  },
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockGeoMap: ApiData<GeoMap> = {
  id: 1,
  name: 'Test Map',
  yearValidity: 2022,
  mesh: 'Test Maille',
  source: 'Test Source',
  license: 'Test License',
  properties: [
    {
      name: 'code',
      sample: '39',
      isGeoCode: true,
    },
    {
      name: 'nom',
      sample: 'Jura',
      isGeoCode: false,
    },
  ],
  owner: mockUser,
  geoJSON: mockGeoJson,
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockTags: ApiData<Tag>[] = [
  {
    id: 1,
    name: 'Tag A',
    count: 5,
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
  {
    id: 2,
    name: 'Tag B',
    count: 3,
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
  {
    id: 3,
    name: 'Tag C',
    count: 2,
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
];

export const mockDataset: ApiData<Dataset> = {
  id: 122,
  name: 'firstDataset',
  origin: 3,
  isPublic: true,
  owner: mockUser,
  csv: {
    id: 5,
    name: 'file.csv',
    hash: 'hashCSV',
    provider: 'provider123',
    ext: '.csv',
    mime: 'application/csv',
    size: 1234,
    url: '/api/v1/Uploads/id/csvFile.csv',
    formats: {
      thumbnail: {
        extension: 'ext',
        url: '/api/v1/Uploads/id/csvFile.csv?720',
        size: 360,
        hash: 'hash720',
        mime: 'mimeExample',
      },
    },
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockDataFragments: ApiData<DataFragment>[] = [
  {
    id: 1,
    name: 'FirstFragment',
    columns: [
      {
        name: 'firstColumn',
        sample: 5,
        source: 'source1',
        validity: '2021',
        isGeoCode: true,
      },
      {
        name: 'secondColumn',
        sample: 10.3,
        source: 'source2',
        validity: '2022',
        isGeoCode: false,
      },
      {
        name: 'thirdColumn',
        sample: 0.25,
        source: 'source3',
        validity: '2023',
        isGeoCode: false,
      },
      {
        name: 'fourthColumn',
        sample: 1000,
        source: 'source4',
        validity: '2024',
        isGeoCode: false,
      },
    ],
    dataset: mockDataset,
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
];

export const indicatorsMockData: ApiData<Indicator>[] = [
  {
    id: 4,
    name: 'mockIndicator1',
    description: 'This is a mock indicator',
    source: 'mock',
    validity: 2,
    created_at: '2023-03-14T10:06:41.979Z',
    updated_at: '2023-03-14T10:06:41.979Z',
    equation: 'A+2',
    variables: [],
  },
  {
    id: 5,
    name: 'mockIndicator2',
    description: 'This is a mock indicator',
    source: 'mock',
    validity: 4,
    variables: [],
    created_at: '2023-03-14T10:12:37.285Z',
    updated_at: '2023-03-14T10:12:37.285Z',
    equation: 'A*2',
  },
];

export const mockPageCartoData: ApiData<PageCarto> = {
  id: 1,
  name: 'Page Carto 1',
  owner: mockUser,
  tags: mockTags,
  map: mockGeoMap,
  cover: mockCover,
  indicators: indicatorsMockData,
  charts: [],
  html: '<p>Test HTML</p>',
  data_fragments: mockDataFragments,
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockGeoMapData: ApiData<GeoMap> = {
  id: 1,
  name: 'Test Map',
  yearValidity: 2022,
  mesh: 'Test Maille',
  source: 'Test Source',
  license: 'Test License',
  properties: [
    {
      name: 'nom',
      sample: 'Taponas',
      isGeoCode: true,
    },
  ],
  owner: mockUser,
  geoJSON: {
    id: 1,
    name: 'test.geojson',
    hash: 'abc123',
    provider: 'provider123',
    ext: '.geojson',
    mime: 'application/json',
    size: 1234,
    url: 'https://example.com/api/v1/geo-maps/id/thumbnail',
    formats: {
      '720': {
        extension: 'ext',
        url: 'happyUrl',
        size: 360,
        hash: 'hashbar',
        mime: 'mimeExample',
      },
    },
    created_at: '2022-02-07T00:00:00.000Z',
    updated_at: '2022-02-07T00:00:00.000Z',
  },
  created_at: '2022-02-07T00:00:00.000Z',
  updated_at: '2022-02-07T00:00:00.000Z',
};

export const mockGeoMapApiCollection: ApiCollection<GeoMap> = {
  data: Array.from({ length: 3 }, (_, index) => ({
    ...mockGeoMapData,
    id: index + 1,
    name: `Test Map ${index + 1}`,
  })),
  meta: {
    pagination: {
      pageCount: 1,
      total: 3,
    },
  },
};
