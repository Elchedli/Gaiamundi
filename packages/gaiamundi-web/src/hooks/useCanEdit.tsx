import { useLocation } from 'react-router-dom';
import { useAuth } from './useAuth';

export const useCanEdit = () => {
  const { isAuthenticated } = useAuth();
  const location = useLocation();
  const isOnEditorPage = location.pathname.endsWith('edit');
  return isAuthenticated && isOnEditorPage;
};
