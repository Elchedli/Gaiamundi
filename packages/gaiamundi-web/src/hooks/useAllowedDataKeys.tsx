import { useMemo } from 'react';
import { ChartType, RawDatumType } from '../interfaces/chart';
import { useChartConfig } from './useChartConfig';
import { useData } from './useData';

export const ALLOWED_DOMAIN_TYPES_BY_CHART: {
  [key in ChartType]: Array<RawDatumType>;
} = {
  column: ['string', 'number'],
  bar: ['string', 'number'],
  scatter: ['number'],
  line: ['number'],
  lineColumn: ['string', 'number'],
  pie: ['number'],
  area: ['number'],
  bubble: ['number'],
};

export const useAllowedDataKeys = () => {
  const { chart: chartConfig } = useChartConfig();
  const { dataKeys } = useData();
  const allowedTypes = ALLOWED_DOMAIN_TYPES_BY_CHART[chartConfig.type];

  const domainKeyOptions = useMemo(
    () =>
      Object.entries(dataKeys)
        .filter(([, type]) => {
          return allowedTypes.indexOf(type) !== -1;
        })
        .map(([key]) => ({
          value: key,
          label: key,
        })),
    [allowedTypes, dataKeys]
  );

  return { domainKeyOptions };
};
