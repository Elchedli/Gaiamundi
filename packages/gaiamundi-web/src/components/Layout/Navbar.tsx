import { PlusIcon } from '@heroicons/react/24/solid';
import { Link, useLocation, useNavigate } from 'react-router-dom';

import classNames from 'classnames';
import { Button } from 'components/Button/Button';
import { AccountDropdown } from 'components/Layout/Account/AccountDropdown';
import { useAuth } from 'hooks/useAuth';
import { LogoLink } from './LogoLink';

const NavbarMenuItem: React.FC<{
  href: string;
  title: string;
  isInverted: boolean;
}> = ({ href, title, isInverted = false }) => {
  const location = useLocation();
  const isCurrentPage = location.pathname === href;
  const lightClasses = {
    active: `text-blue-900 bg-blue-200 focus:text-blue-600`,
    inactive: `text-blue-600 hover:text-blue-900 hover:bg-blue-100 focus:outline-none`,
  };
  const darkClasses = {
    active: `text-white bg-royal-blue-800 focus:text-white`,
    inactive: `text-blue-100 hover:text-white hover:bg-royal-blue-800 focus:outline-none`,
  };
  const classes = isInverted ? darkClasses : lightClasses;
  return (
    <Link
      to={href}
      className={`mr-4 px-3 py-2 rounded text-sm font-medium focus:outline-none ${
        isCurrentPage ? classes.active : classes.inactive
      }`}
    >
      {title}
    </Link>
  );
};

export const libMenuItems = [
  { title: 'Démarrer', href: '#' },
  { title: 'Documentation', href: '#' },
  { title: 'Cartes', href: '#' },
  { title: 'Blog', href: '#' },
];

const loggedInMenuItems = [
  { title: 'Tableau de bord', href: '/dashboard' },
  { title: 'Compte', href: '/account' },
];

const anonymousMenuItems = [
  { title: 'Fonctionnalités', href: '/#features' },
  { title: 'Documentation', href: '/#docs' },
  { title: 'Equipe', href: '/#team' },
  { title: 'Blog', href: '/blog' },
];

const AuthSidebarMenu = (): JSX.Element => {
  return (
    <div data-testid="login-buttons">
      <Link to="/login">
        <Button
          color="lime"
          outline={false}
          size="md"
          className="inline-flex mx-1"
        >
          Se connecter
        </Button>
      </Link>
      <Link to="/signup">
        <Button
          gradientMonochrome="blue"
          outline={false}
          size="md"
          className="inline-flex mx-1"
        >
          Créer un compte
        </Button>
      </Link>
    </div>
  );
};

const NavbarMenu: React.FC<{ isInverted: boolean }> = ({ isInverted }) => {
  const { isAuthenticated } = useAuth();
  const location = useLocation();
  const isHomePage = location.pathname === '/';
  const menuItems =
    isAuthenticated && !isHomePage ? loggedInMenuItems : anonymousMenuItems;
  return (
    <div data-testid="navigations" className="flex items-baseline ml-10">
      {menuItems.map(({ title, href }) => {
        return (
          <NavbarMenuItem
            key={title}
            href={href}
            title={title}
            isInverted={isInverted}
          />
        );
      })}
    </div>
  );
};

const SideNavigation: React.FC = () => {
  const navigate = useNavigate();
  const { logout } = useAuth();
  const isAppPage = location.pathname.startsWith('/app');
  return (
    <div className="flex flex-row items-center hidden md:flex">
      {!isAppPage && (
        <div className="flex flex-row">
          <Button
            outline={false}
            size="md"
            className="inline-flex mx-1"
            color={'lime'}
            icon={PlusIcon}
            data-testid="newPageCarto-button"
            onClick={() => navigate('/page-carto/create')}
          >
            Nouvelle PageCarto
          </Button>
          <Button
            outline={false}
            size="md"
            className="inline-flex mx-1"
            color={'blue'}
            onClick={() => logout()}
          >
            Déconnexion
          </Button>
        </div>
      )}
      <AccountDropdown />
    </div>
  );
};

type NavbarProps = {
  isFluid?: boolean;
  isInverted?: boolean;
};

const Navbar: React.FC<NavbarProps> = ({
  isFluid = false,
  isInverted = false,
}) => {
  const { isAuthenticated } = useAuth();
  return (
    <nav
      className={classNames(
        isFluid ? 'w-full' : '',
        isInverted ? 'bg-royal-blue-900 text-gray-700 body-font' : 'bg-white',
        'shadow-md z-10'
      )}
    >
      <div className={`sm:px-6 lg:px-8 ${!isFluid && 'mx-auto max-w-7xl'}`}>
        <div className="">
          <div className="flex items-center justify-between h-16 px-4 sm:px-0">
            <div className="flex items-center">
              <LogoLink isInverted={isInverted} />
              <NavbarMenu isInverted={isInverted} />
            </div>
            {isAuthenticated ? <SideNavigation /> : <AuthSidebarMenu />}
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
