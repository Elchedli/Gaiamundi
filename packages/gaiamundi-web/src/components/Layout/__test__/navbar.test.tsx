import { fireEvent, render, screen } from '@testing-library/react';
import Navbar from 'components/Layout/Navbar';
import { useAuth } from 'hooks/useAuth';
import { MemoryRouter as Router } from 'react-router-dom';
import { mockUser } from 'utils/mocks/data';

jest.mock('hooks/useAuth');

describe('Navbar', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render authentication menu when not authenticated', () => {
    (useAuth as jest.Mock).mockImplementation(() => ({
      isAuthenticated: false,
    }));
    const { container } = render(
      <Router>
        <Navbar />
      </Router>
    );

    const loginButtons = screen.getByTestId('loginButtons').childNodes;
    expect(loginButtons[0]).toHaveAttribute('href', '/login');
    expect(loginButtons[1]).toHaveAttribute('href', '/signup');
    expect(container).toMatchSnapshot();
  });

  it('should render logged-in menu when authenticated and not on home page', () => {
    (useAuth as jest.Mock).mockImplementation(() => ({
      isAuthenticated: true,
      user: mockUser,
    }));

    const { container } = render(
      <Router>
        <Navbar />
      </Router>
    );

    expect(screen.getByTestId('navigations').childNodes).toHaveLength(4);
    fireEvent.click(screen.getAllByTestId('button-element')[0]);
    expect(screen.getByTestId('navigations').childNodes).toHaveLength(2);
    expect(container).toMatchSnapshot();
  });
});
