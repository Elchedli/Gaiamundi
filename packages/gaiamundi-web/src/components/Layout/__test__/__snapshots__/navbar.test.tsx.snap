// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render authentication menu when not authenticated 1`] = `
<div>
  <nav
    class="bg-white shadow-md z-10"
  >
    <div
      class="sm:px-6 lg:px-8 mx-auto max-w-7xl"
    >
      <div
        class=""
      >
        <div
          class="flex items-center justify-between h-16 px-4 sm:px-0"
        >
          <div
            class="flex items-center"
          >
            <a
              href="/"
            >
              <svg
                height="38"
                viewBox="0 0 400 76"
                width="200"
              >
                <defs
                  id="defs34"
                >
                  <rect
                    height="35.97667"
                    id="rect248"
                    width="163.26823"
                    x="35.374524"
                    y="1.1788015"
                  />
                  <rect
                    height="34.610613"
                    id="rect248-7"
                    width="28.96704"
                    x="35.374523"
                    y="1.1788015"
                  />
                </defs>
                <text
                  id="text246"
                  style="font-size: 26.6667px; white-space: pre; fill: #000000;"
                  transform="matrix(2.1768629,0,0,2.1768629,-3.6143566,4.8736603)"
                >
                  <tspan
                    id="tspan499"
                    x="35.375"
                    y="24.773467"
                  >
                    <tspan
                      id="tspan497"
                      style="font-family: DejaVu Serif;"
                    >
                      Gaïamundi
                    </tspan>
                  </tspan>
                </text>
                <g
                  id="Layer_1"
                  transform="matrix(2.9937644,0,0,2.9937644,-0.70475605,2.323416)"
                >
                  <g
                    id="g410"
                  >
                    <path
                      d="m 12,5 c 3.859,0.001 7,3.142 7,7.001 C 19,15.859 15.859,18.999 12,19 8.141,19 5,15.86 5,12.001 5,8.142 8.141,5.001 12,5 M 12,3 C 7.029,3.001 3,7.03 3,12.001 3,16.971 7.029,21 12,21 16.97,20.999 21,16.97 21,12.001 21,7.03 16.971,3.001 12,3 Z"
                      id="path408"
                    />
                  </g>
                  <path
                    d="M 16.182,7.819 C 16.053,7.691 15.867,7.641 15.691,7.692 L 9.74,9.398 C 9.574,9.446 9.445,9.575 9.398,9.741 l -1.707,5.951 c -0.051,0.175 -0.002,0.363 0.127,0.491 0.095,0.095 0.223,0.146 0.354,0.146 0.045,0 0.092,-0.006 0.138,-0.02 l 5.95,-1.708 c 0.165,-0.047 0.295,-0.177 0.342,-0.343 l 1.707,-5.949 c 0.05,-0.173 0.002,-0.361 -0.127,-0.49 z m -7.282,7.282 1.383,-4.817 3.434,3.435 z"
                    id="path412"
                  />
                </g>
              </svg>
            </a>
            <div
              class="flex items-baseline ml-10"
              data-testid="navigations"
            >
              <a
                class="mr-4 px-3 py-2 rounded text-sm font-medium focus:outline-none text-gray-600 hover:text-gray-900 hover:bg-gray-100 focus:outline-none"
                href="/#features"
              >
                Fonctionnalités
              </a>
              <a
                class="mr-4 px-3 py-2 rounded text-sm font-medium focus:outline-none text-gray-600 hover:text-gray-900 hover:bg-gray-100 focus:outline-none"
                href="/#docs"
              >
                Documentation
              </a>
              <a
                class="mr-4 px-3 py-2 rounded text-sm font-medium focus:outline-none text-gray-600 hover:text-gray-900 hover:bg-gray-100 focus:outline-none"
                href="/#team"
              >
                Equipe
              </a>
              <a
                class="mr-4 px-3 py-2 rounded text-sm font-medium focus:outline-none text-gray-600 hover:text-gray-900 hover:bg-gray-100 focus:outline-none"
                href="/blog"
              >
                Blog
              </a>
            </div>
          </div>
          <div
            data-testid="loginButtons"
          >
            <a
              href="/login"
            >
              <button
                class="group inline-flex h-min w-fit items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg bg-lime-300 border border-transparent hover:bg-lime-400 focus:ring-4 focus:ring-lime-300 disabled:hover:bg-lime-700 dark:bg-lime-600 dark:hover:bg-lime-700 dark:focus:ring-lime-800 dark:disabled:hover:bg-lime-600 inline-flex mx-1"
                data-testid="button-element"
                type="button"
              >
                <span
                  class="flex items-center text-sm px-4 py-2 rounded-md"
                >
                  Se connecter
                </span>
              </button>
            </a>
            <a
              href="/signup"
            >
              <button
                class="group inline-flex h-min w-fit items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg text-white bg-gradient-to-r from-blue-500 via-blue-600 to-blue-700 hover:bg-gradient-to-br focus:ring-4 focus:ring-blue-300 dark:focus:ring-blue-800  inline-flex mx-1"
                data-testid="button-element"
                type="button"
              >
                <span
                  class="flex items-center text-sm px-4 py-2 rounded-md"
                >
                  Créer un compte
                </span>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Navbar should render logged-in menu when authenticated and not on home page 1`] = `
<div>
  <nav
    class="bg-white shadow-md z-10"
  >
    <div
      class="sm:px-6 lg:px-8 mx-auto max-w-7xl"
    >
      <div
        class=""
      >
        <div
          class="flex items-center justify-between h-16 px-4 sm:px-0"
        >
          <div
            class="flex items-center"
          >
            <a
              href="/"
            >
              <svg
                height="38"
                viewBox="0 0 400 76"
                width="200"
              >
                <defs
                  id="defs34"
                >
                  <rect
                    height="35.97667"
                    id="rect248"
                    width="163.26823"
                    x="35.374524"
                    y="1.1788015"
                  />
                  <rect
                    height="34.610613"
                    id="rect248-7"
                    width="28.96704"
                    x="35.374523"
                    y="1.1788015"
                  />
                </defs>
                <text
                  id="text246"
                  style="font-size: 26.6667px; white-space: pre; fill: #000000;"
                  transform="matrix(2.1768629,0,0,2.1768629,-3.6143566,4.8736603)"
                >
                  <tspan
                    id="tspan499"
                    x="35.375"
                    y="24.773467"
                  >
                    <tspan
                      id="tspan497"
                      style="font-family: DejaVu Serif;"
                    >
                      Gaïamundi
                    </tspan>
                  </tspan>
                </text>
                <g
                  id="Layer_1"
                  transform="matrix(2.9937644,0,0,2.9937644,-0.70475605,2.323416)"
                >
                  <g
                    id="g410"
                  >
                    <path
                      d="m 12,5 c 3.859,0.001 7,3.142 7,7.001 C 19,15.859 15.859,18.999 12,19 8.141,19 5,15.86 5,12.001 5,8.142 8.141,5.001 12,5 M 12,3 C 7.029,3.001 3,7.03 3,12.001 3,16.971 7.029,21 12,21 16.97,20.999 21,16.97 21,12.001 21,7.03 16.971,3.001 12,3 Z"
                      id="path408"
                    />
                  </g>
                  <path
                    d="M 16.182,7.819 C 16.053,7.691 15.867,7.641 15.691,7.692 L 9.74,9.398 C 9.574,9.446 9.445,9.575 9.398,9.741 l -1.707,5.951 c -0.051,0.175 -0.002,0.363 0.127,0.491 0.095,0.095 0.223,0.146 0.354,0.146 0.045,0 0.092,-0.006 0.138,-0.02 l 5.95,-1.708 c 0.165,-0.047 0.295,-0.177 0.342,-0.343 l 1.707,-5.949 c 0.05,-0.173 0.002,-0.361 -0.127,-0.49 z m -7.282,7.282 1.383,-4.817 3.434,3.435 z"
                    id="path412"
                  />
                </g>
              </svg>
            </a>
            <div
              class="flex items-baseline ml-10"
              data-testid="navigations"
            >
              <a
                class="mr-4 px-3 py-2 rounded text-sm font-medium focus:outline-none text-gray-600 hover:text-gray-900 hover:bg-gray-100 focus:outline-none"
                href="/dashboard"
              >
                Tableau de bord
              </a>
              <a
                class="mr-4 px-3 py-2 rounded text-sm font-medium focus:outline-none text-gray-600 hover:text-gray-900 hover:bg-gray-100 focus:outline-none"
                href="/account"
              >
                Compte
              </a>
            </div>
          </div>
          <div
            class="flex flex-row items-center hidden md:flex"
          >
            <div
              class="flex flex-row"
            >
              <button
                class="group inline-flex h-min w-fit items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg bg-lime-300 border border-transparent hover:bg-lime-400 focus:ring-4 focus:ring-lime-300 disabled:hover:bg-lime-700 dark:bg-lime-600 dark:hover:bg-lime-700 dark:focus:ring-lime-800 dark:disabled:hover:bg-lime-600 inline-flex mx-1"
                data-testid="button-element"
                type="button"
              >
                <span
                  class="flex items-center text-sm px-4 py-2 rounded-md !px-2"
                >
                  <svg
                    aria-hidden="true"
                    class="w-5 h-5"
                    fill="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 3.75a.75.75 0 01.75.75v6.75h6.75a.75.75 0 010 1.5h-6.75v6.75a.75.75 0 01-1.5 0v-6.75H4.5a.75.75 0 010-1.5h6.75V4.5a.75.75 0 01.75-.75z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  Nouvelle PageCarto
                </span>
              </button>
              <button
                class="group inline-flex h-min w-fit items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg text-white bg-blue-700 border border-transparent hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 disabled:hover:bg-blue-700 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 dark:disabled:hover:bg-blue-600 inline-flex mx-1"
                data-testid="button-element"
                type="button"
              >
                <span
                  class="flex items-center text-sm px-4 py-2 rounded-md"
                >
                  Déconnexion
                </span>
              </button>
            </div>
            <div
              class="flex items-center ml-4 md:ml-6"
            >
              <div
                class="relative ml-3"
              >
                <button
                  class="flex items-center max-w-xs text-sm text-white rounded-full focus:outline-none focus:shadow-solid"
                >
                  <span
                    class="inline-block w-8 h-8 overflow-hidden bg-gray-200 rounded-full"
                  >
                    <svg
                      class="w-8 h-8"
                      fill="currentColor"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M24 20.993V24H0v-2.996A14.977 14.977 0 0112.004 15c4.904 0 9.26 2.354 11.996 5.993zM16.002 8.999a4 4 0 11-8 0 4 4 0 018 0z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
